relay-server/src/extractors/shared_payload.rs

```diff
@@ -50,18 +53,17 @@ impl Stream for SharedPayload {
 
     #[inline]
     fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
-        if self.done {
+        if self.done.load(Ordering::Relaxed) {
+            relay_log::info!("done");
             return Ok(Async::Ready(None));
         }
 
         let poll = self.inner.poll();
 
-        self.done = match poll {
```

Comment:
This was overwriting `done` in one clone of the shared payload, but not the others.